instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/rum/RumConfigE2ETests.kt

```diff
@@ -668,6 +674,108 @@ class RumConfigE2ETests {
         sendAllRumEvents(forge, testMethodName)
     }
 
+    /**
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun setVitalsUpdateFrequency(VitalsUpdateFrequency): Builder
+     */
+    @Test
+    fun rum_config_set_vitals_update_frequency_never() {
+        val testMethodName = "rum_config_set_vitals_update_frequency_never"
+        val strategy = ActivityViewTrackingStrategy(
+            true,
+            componentPredicate = object : ComponentPredicate<Activity> {
+                override fun accept(component: Activity): Boolean {
+                    return true
+                }
+
+                override fun getViewName(component: Activity): String {
+                    return testMethodName
+                }
+            }
+        )
+        // we need to initialize the SDK on the Main Thread (Looper thread) otherwise the
+        // Choreographer Callback will not be registered
+        InstrumentationRegistry.getInstrumentation().runOnMainSync {
+            measureSdkInitialize {
+                val config = Configuration
+                    .Builder(
+                        logsEnabled = true,
+                        tracesEnabled = true,
+                        rumEnabled = true,
+                        crashReportsEnabled = true
+                    )
+                    .useViewTrackingStrategy(strategy)
+                    .setVitalsUpdateFrequency(VitalsUpdateFrequency.NEVER)
+                    .build()
+                initializeSdk(
+                    InstrumentationRegistry.getInstrumentation().targetContext,
+                    TrackingConsent.GRANTED,
+                    config
+                )
+            }
+        }
+        val scenario = launch(ViewTrackingActivity::class.java)
+        // Give some time to vitals monitors to be updated
+        Thread.sleep(5000)
+        scenario.onActivity {
+            InstrumentationRegistry.getInstrumentation().callActivityOnPause(it)
+            InstrumentationRegistry.getInstrumentation().callActivityOnResume(it)
+        }
+        InstrumentationRegistry.getInstrumentation().waitForIdleSync()
+    }
+
+    /**
+     * apiMethodSignature: com.datadog.android.core.configuration.Configuration$Builder#fun setVitalsUpdateFrequency(VitalsUpdateFrequency): Builder
+     */
+    @Test
+    fun rum_config_set_vitals_update_frequency() {
+        val fakeFrequency = forge.aValueFrom(
+            VitalsUpdateFrequency::class.java,
+            exclude = listOf(VitalsUpdateFrequency.NEVER)
+        )
+        val testMethodName = "rum_config_set_vitals_update_frequency"
+        // we need to initialize the SDK on the Main Thread (Looper thread) otherwise the
```

Comment:
very minor: this comment belongs to L750.

Reply:
@0xnm regarding the choreographer...you are not wrong, for some reason during my tests sometimes it was not triggered if I didn't play a bit with the activity.

Reply:
> very minor: this comment belongs to L750.

yeah, I think it was messed up after I did the refactor by extracting the variable.